crates/rpc/rpc/src/eth/cache.rs

```diff
@@ -380,6 +380,41 @@ where
     }
 }
 
+impl<Client, Tasks> EthStateCacheService<Client, Tasks>
+where
+    Client: StateProviderFactory + BlockProvider + EvmEnvProvider + Clone + Unpin + 'static,
+    Tasks: TaskSpawner + Clone + 'static,
+{
+    /// Check for new blocks and insert them in cache
+    async fn check_for_new_blocks<St>(&mut self, mut events: St)
+    where
+        St: Stream<Item = CanonStateNotification> + Unpin + 'static,
```

Comment:
please make this a standalone function.

this also needs to operate on the `EthStateCache` and not `EthStateCacheService`